tests/e2e_test.go

```diff
@@ -294,15 +315,205 @@ var _ = ginkgo.BeforeSuite(func() {
 	}
 	log.Info("Finished deploying Teleporter contracts")
 
+	log.Info("Starting relayer goroutine A->B")
+	go func() {
+		ctx := context.Background()
+		subA, err := chainAWSClient.SubscribeNewHead(ctx, newHeadsA)
+		Expect(err).Should(BeNil())
+		defer subA.Unsubscribe()
+
+		// Get the latest block from Subnet A, and retrieve the warp message from the logs
+		log.Info("Waiting for new block confirmation")
+		for newHeadA := range newHeadsA {
+			blockHashA := newHeadA.Hash()
+
+			log.Info("Fetching relevant warp logs from the newly produced block")
+			logs, err := chainARPCClient.FilterLogs(ctx, interfaces.FilterQuery{
+				BlockHash: &blockHashA,
+				Addresses: []common.Address{warp.Module.Address},
+			})
+			Expect(err).Should(BeNil())
+			Expect(len(logs)).Should(Equal(1))
+
+			// Check for relevant warp log from subscription and ensure that it matches
+			// the log extracted from the last block.
+			txLog := logs[0]
+			log.Info("Parsing logData as unsigned warp message")
+			unsignedMsg, err := avalancheWarp.ParseUnsignedMessage(txLog.Data)
+			Expect(err).Should(BeNil())
+
+			// Set local variables for the duration of the test
+			unsignedWarpMessageID := unsignedMsg.ID()
+			unsignedWarpMsg := unsignedMsg
+			log.Info("Parsed unsignedWarpMsg", "unsignedWarpMessageID", unsignedWarpMessageID, "unsignedWarpMessage", unsignedWarpMsg)
+
+			// Loop over each client on chain A to ensure they all have time to accept the block.
+			// Note: if we did not confirm this here, the next stage could be racy since it assumes every node
+			// has accepted the block.
+			for i, uri := range chainANodeURIs {
+				chainAWSURI := httpToWebsocketURI(uri, blockchainIDA.String())
+				log.Info("Creating ethclient for blockchainA", "wsURI", chainAWSURI)
+				client, err := ethclient.Dial(chainAWSURI)
```

Comment:
We're dialing the client an arbitrary number of times here. We should either dial once and save the client, or we should close the connection after every loop.